src/hist/core.py

```diff
@@ -2,29 +2,270 @@
 import matplotlib
 import matplotlib.pyplot as plt
 import matplotlib.patches as patches
+from matplotlib import transforms
 from scipy.optimize import curve_fit
 from uncertainties import correlated_values, unumpy
 from boost_histogram import Histogram
-from typing import Callable, Optional, Tuple
+from typing import Callable, Optional, Tuple, Union
 
 
 class BaseHist(Histogram):
     def __init__(self, *args, **kwargs):
         """
-            Initialize Hist object. Axis params must contain the names.
+            Initialize BaseHist object. Axis params can contain the names.
         """
 
         super().__init__(*args, **kwargs)
         self.names: dict = dict()
         for ax in self.axes:
             if ax.name in self.names:
                 raise Exception(
-                    "Hist instance cannot contain axes with duplicated names."
+                    f"{self.__class__.__name__} instance cannot contain axes with duplicated names."
                 )
             else:
                 self.names[ax.name] = True
 
-    def pull_plot(
+    def project(self, *args: Union[int, str]):
+        """
+        Projection of axis idx.
+        """
+        if len(args) == 0 or all(isinstance(x, int) for x in args):
+            return super().project(*args)
+
+        elif all(isinstance(x, str) for x in args):
+            indices: tuple = tuple()
+            for name in args:
+                for index, axis in enumerate(self.axes):
+                    if name == axis.name:
+                        indices += (index,)
+                        break
+                else:
+                    raise ValueError("The axis names could not be found")
+
+            return super().project(*indices)
+
+        else:
+            raise TypeError(
+                f"Only projections by indices and names are supported for {self.__class__.__name__}"
+            )
+
+    def plot(
+        self, *args, **kwargs,
+    ):  # ToDo: how to add return type hints
```

Comment:
It's just a Union of the other two. I'd make a type alias, such as:

```python
Plot1D_RetType = Tuple[matplotlib.figure.Figure, matplotlib.axes._subplots.SubplotBase]
```

Then use it here in a `Union[Plot1D_RetType, Plot2D_RetType]`, as well as for the 1D return (also 2D).

Reply:
Okay, it works fine. But I wonder why I cannot use like this:

```python
Fig_Type = matplotlib.figure.Figure
Ax_Type = matplotlib.axes._subplots.SubplotBase
Plot2D_RetType = Tuple[Fig_Type, Ax_Type]
```

Looks like `Fig_Type` is not a **type** :)